### DISCUSSION TITLE
Change ports that transparent mode redirects to (i.e 80 and 443)

### DISCUSSION THREAD

**Discussion by Botytec** [mitmproxy/mitmproxy#6263] [General]
when i use transparent mode it only works on 80 and 443, is there a way to change or add a custom port?

**Comment by mhils** [comment]
Can you clarify what exactly you are doing? Which OS are you on?

**Reply by Botytec** [reply]
I am on windows and i am trying to use mitm proxy as a transparent proxy

**Reply by mhils** [reply]
You intend to proxy the traffic from your Windows machine itself, or from another device?

**Reply by Botytec** [reply]
my windows machine

**Reply by mhils** [reply]
Ok. So we do have some new stuff coming up here very soon, but it's still completely undocumented. In a nutshell:

1. Go to https://github.com/mitmproxy/mitmproxy/actions/workflows/main.yml?query=branch%3Amain click on the latest build and download the binary artifacts for Windows. Install mitmproxy from there.
2. Run `mitmweb --mode osproxy` to intercept all traffic from your device (mitmdump/mitmproxy work just as well).
You can also run `mitmweb --mode osproxy:curl.exe` to only intercept a specific application (cURL in this case).
3. Report back any issues, this is very much in development! 😃

**Reply by Botytec** [reply]
i cant run it in transparent mode though? or is osproxy the same thing

**Reply by mhils** [reply]
osproxy mode is "transparently intercept all traffic from the current device". So yes. :)

**Reply by Botytec** [reply]
can i also ask you another question sir? im trying to decrypt this data to make a server emulator of a game but it uses certificate pinning, when the server askes for the certificate, the proxy sends no certificate. Is there away to like allow the client to send its certificate thru while im proxying so the server can trust me and allow me to connect but also have my proxy on?

**Reply by mhils** [reply]
You would need to configure mitmproxy to use a client certificate. For that you need the actual certificate first.

**Reply by Botytec** [reply]
Would I need just the certificate or the private key also?

**Reply by mhils** [reply]
Private Key as well.
